SeaPkg/Library/BasePeCoffValidationLib/Validation.c

```diff
@@ -0,0 +1,325 @@
+/** @file
+  Defines function definitions for validating a loaded PE/COFF image post execution
+  against an auxiliary file (defined in SeaAuxiliary.h) that contains auxiliary
+  multiple validation entries. A validation entry specifies a specific region in the
+  target image buffer to validate against, a validation type to perform, and any
+  necessary data to assist in the validation.
+
+  There currently exists five validation types:
+  - Non-Zero: Validates that the specified region in the target image buffer is not all zero.
+  - Content: Validates that the specified region in the target image buffer matches the content in the reference data.
+  - MemAttr: Validates that the specified region in the target image buffer belongs to a user page that is mapped inside MM
+             and the page attributes match the requirements specified by the validation entry.
+  - SelfRef: Validates that the specified region in the target image buffer matches the content in the original image buffer as
+             specified by TargetOffset in the validation entry.
+  - Pointer: Validates that the specified region in the target image buffer is a pointer, and that the pointer is not NULL.
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+#include <Uefi.h>
+#include <Base.h>
+#include <SeaAuxiliary.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/DebugLib.h>
+#include <Library/PeCoffValidationLib.h>
+#include <Library/SafeIntLib.h>
+
+EFI_STATUS
+EFIAPI
+GetMemoryAttributes (
+  IN  EFI_PHYSICAL_ADDRESS  PageTableBase,
+  IN  EFI_PHYSICAL_ADDRESS  BaseAddress,
+  IN  UINT64                Length,
+  OUT UINT64                *Attributes
+  );
+
+/**
+  Helper function that will evaluate the page where the input address is located belongs to a
+  user page that is mapped inside MM.
+  @param  Address           Target address to be inspected.
+  @param  Size              Address range to be inspected.
+  @param  IsUserRange       Pointer to hold inspection result, TRUE if the region is in User pages, FALSE if
+                            the page is in supervisor pages. Should not be used if return value is not EFI_SUCCESS.
+  @return     The result of inspection operation.
+**/
+EFI_STATUS
+InspectTargetRangeAttribute (
+  IN  EFI_PHYSICAL_ADDRESS  PageTableBase,

```

Comment:
The params in the function description comment currently don't match the actual params 1:1.